lib/api/CDataFrameTrainBoostedTreeRunner.cc

```diff
@@ -357,20 +359,26 @@ void CDataFrameTrainBoostedTreeRunner::runImpl(core::CDataFrame& frame) {
             treeRestored = this->restoreBoostedTree(frame, dependentVariableColumn,
                                                     restoreSearcher);
         }
-        if (treeRestored == false) {
+        if (treeRestored == false && m_Task == E_Train) {
             m_BoostedTree = m_BoostedTreeFactory->buildForTrain(frame, dependentVariableColumn);
         }
+        if (m_Task == E_Predict) {
+            m_BoostedTree = m_BoostedTreeFactory->buildForPredict(frame, dependentVariableColumn);
+        }
```

Comment:
Can these setup tasks move into the switch statements below (seems cleaner if they can), or do they need to precede validate?